### ISSUE TITLE
Doesn't find method headers if there's an annotation

### ISSUE THREAD

**Issue by purag** [anikan/JavaStyleChecker#3]
For example, `@Override` on its own line breaks the method header check.

**Comment by anikan** [comment]
Interesting, can you provide an example file?

**Comment by purag** [comment]
```
/* Here's a method header.
 * Some comments.
 * Some more
 * close it off now.
 */
@Override
public String someMethod(){
}
```

Also fails when `@Override` is on the same line.

**Comment by anikan** [comment]
The way I search for method headers is first finding the name of the method (the part just before the parentheses) and then searching for a line that has "name: <methodName>", not sure if there's a better way of doing that. 

In your example, you don't have that last line. Could that be the issue? Or are the annotations messing up the regex?

**Comment by purag** [comment]
I think that works for now, but maybe in the future we should just look for a `/*` and `*/` near method name. I don't think we should search for specific content because people all do method headers differently.

**Comment by purag** [comment]
I ran the script when grading and it couldn't find a method header; now that I'm looking back, it didn't contain "name: ".

**Comment by anikan** [comment]
Yeah it's tricky because some people write the method names in regular multiline comments, like 

/*
*Using method to do something
*/

and that isn't a header.
